### PR TITLE
debt: Remove code related to user public repos

### PR DESCRIPTION
**Pull request by ryanslade**
This was a concept we had in Cloud V1 which we no longer support.

Part of https://github.com/sourcegraph/sourcegraph/issues/44700

## Test plan

Existing tests

### CODE REVIEW

**Comment by ryanslade** [comment]
That felt good 🔥

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (0.00 kb) | 0.00% (0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/rs/remove-user-public-repos-code/compare-report.html;tab=live_object) for a full comparison between the commits db91f65 and f6b8985 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 1]
```diff
@@ -5471,7 +5470,7 @@ type User implements Node & SettingsSubject & Namespace {
     namespaceName: String!
 
     """
-    publicRepositories returns the repos listed in user_public_repos for this user
+    DEPRECATED
     """
     publicRepositories: [Repository!]!
```
Comment (mrnugget) [inline cmd/frontend/graphqlbackend/schema.graphql:5475]: I think you can just remove it. I'm 100% sure dotcom is the only client on which this ever worked, so we can remove it.

File: cmd/frontend/graphqlbackend/set_user_public_repos.go [hunk 2]
```diff
@@ -2,125 +2,15 @@ package graphqlbackend
...
-
-	// it was able to find a repo, but annoyingly doesn't return the repo ID.
-	// getting the repo ID out of RepoLookup is non-trivial, so instead we'll
-	// just look the repo up by name. janky.
-	repo, err = repoStore.GetByName(ctx, repoName)
-	if err != nil {
-		return nil, errors.Wrap(err, "couldn't find repo after fetching from code host")
-	}
-	return repo, err
+	return nil, errors.Errorf("SetUserPublicRepos has been deprecated")
```
Comment (mrnugget) [inline cmd/frontend/graphqlbackend/set_user_public_repos.go:15]: That's the same as removing it, so feel free to remove it :)

**Review by mrnugget — APPROVED** [review]
VERY NICE!!!

File: cmd/frontend/graphqlbackend/schema.graphql (hunk 1 above)
Comment (eseliger) [reply cmd/frontend/graphqlbackend/schema.graphql:5475]: Yep, agreed.

**Review by eseliger — APPROVED** [review]
can we also drop the DB table?

**Comment by ryanslade** [comment]
> can we also drop the DB table?

We can do that in a separate PR. I know @mrnugget was concerned about waiting until all code had been removed before finally changing the DB schema.

**Review by sashaostrikov — COMMENTED** [review]
NICE!

**Pull request merged**